File: src/main/java/io/github/dqualizer/dqlang/types/adapter/options/RampingScenario.java
package io.github.dqualizer.dqlang.types.adapter.options;

import com.fasterxml.jackson.annotation.JsonTypeName;
import io.github.dqualizer.dqlang.types.adapter.k6.Stage;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.LinkedHashSet;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@JsonTypeName("ramping-vus")
public class RampingScenario extends Scenario {

    private final String executor = "ramping-vus";
    private final int startVUs = 0;
    private LinkedHashSet<Stage> stages;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/k6/options/RampingScenario.kt
package io.github.dqualizer.dqlang.types.adapter.k6.options

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonTypeName("ramping-vus")
data class RampingScenario @JsonCreator constructor(
    val executor: String = "ramping-vus",

    val startVUs: Int = 0,

    val stages: Set<Stage>? = HashSet()
) : Scenario()

